Python/Product/Analysis/ProjectEntry.cs

```diff
@@ -144,8 +144,8 @@ private class ActivePythonParse : IPythonParse {
             }
         }
 
-        public IPythonParse WaitForCurrentParse(int timeout = -1) {
-            if (!_pendingParse.Wait(timeout)) {
+        public IPythonParse WaitForCurrentParse(int timeout, CancellationToken token) {
```

Comment:
Can we make this method async?

Reply:
I'd rather not. This is only ever used in contexts where blocking briefly (and it's only a parse, so it's brief) is the right action.

Reply:
How many times `GetCompletions` can be called while we're waiting for a parse?